### ISSUE TITLE
Link doesn't take styling properly

### ISSUE THREAD

**Issue by jasonrclark** [shoes/shoes4#1136]
Noticed during #1134 but doesn't actually have to do with hovering.

This app, if you comment in/out various styling lines, will give you a variety of different crashes:

```
Shoes.app do
  pa = para("Text", l = link("yO"))

  l.size = 42
  l.stroke = blue
  l.fill = yellow
end
```

`size=` gives this:

```
AfterDo::CallbackError: A NoMethodError: undefined method `element_left' for (Shoes::Link:0x00003eb4 "yO"):Shoes::Link was raised during an after_do callback block for method 'update_style' on the instance Shoes::Common::Style with the following arguments: {:size=>42} defined in the file /Users/jclark/source/shoes4/shoes-swt/lib/shoes/swt/redrawing_aspect.rb in line 81. This is the backtrace of the NoMethodError:
 /Users/jclark/source/shoes4/shoes-swt/lib/shoes/swt/redrawing_aspect.rb:99:in `redraw_element'
/Users/jclark/source/shoes4/shoes-swt/lib/shoes/swt/redrawing_aspect.rb:83:in `add_redraws'
org/jruby/RubyProc.java:271:in `call'
/Users/jclark/.rbenv/versions/jruby-1.7.16.1/lib/ruby/gems/shared/gems/after_do-0.3.1/lib/after_do.rb:113:in `_after_do_execute_callback'
/Users/jclark/.rbenv/versions/jruby-1.7.16.1/lib/ruby/gems/shared/gems/after_do-0.3.1/lib/after_do.rb:107:in `_after_do_execute_callbacks'
org/jruby/RubyArray.java:1613:in `each'
/Users/jclark/.rbenv/versions/jruby-1.7.16.1/lib/ruby/gems/shared/gems/after_do-0.3.1/lib/after_do.rb:106:in `_after_do_execute_callbacks'
/Users/jclark/.rbenv/versions/jruby-1.7.16.1/lib/ruby/gems/shared/gems/after_do-0.3.1/lib/after_do.rb:100:in `update_style'
/Users/jclark/source/shoes4/shoes-core/lib/shoes/common/style.rb:30:in `style'
/Users/jclark/source/shoes4/shoes-core/lib/shoes/common/style.rb:103:in `size='
```

Where `stroke=` gives this:

```
NameError: undefined local variable or method `update_stroke' for (Shoes::Link:0x00003eb4 "yO"):Shoes::Link
            set_coloring at /Users/jclark/source/shoes4/shoes-core/lib/shoes/common/style.rb:145
            update_style at /Users/jclark/source/shoes4/shoes-core/lib/shoes/common/style.rb:125
            update_style at /Users/jclark/.rbenv/versions/jruby-1.7.16.1/lib/ruby/gems/shared/gems/after_do-0.3.1/lib/after_do.rb:99
                   style at /Users/jclark/source/shoes4/shoes-core/lib/shoes/common/style.rb:30
                 stroke= at /Users/jclark/source/shoes4/shoes-core/lib/shoes/common/style.rb:103
                  (root) at /Users/jclark/source/shoes4/testing/link2.rb:5
```

These all seem to be related to the fact that a `Link` isn't really a full Shoes element, but it tries to pretend like it is one. Styling is presently beyond its limits, but shouldn't be.

**Comment by KCErb** [comment]
This sounds familiar to me. I remember running into something like this back when, but I'm not sure if I logged anything in those extremely long conversations or not . . .

**Comment by jasonrclark** [comment]
Searched and couldn't find another direct issue. Probably locked somewhere in conversation. ¯\_(ツ)_/¯

**Comment by PragTob** [comment]
Is this fixed with #1281 now? :)

**Comment by jasonrclark** [comment]
Yep, fixed!
